src/s3/util/fileViewerManager.ts

```diff
@@ -2,30 +2,237 @@
  * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import { OutputChannel } from 'vscode'
 import { ext } from '../../shared/extensionGlobals'
-import { S3FileNode } from '../explorer/s3FileNode'
-import { Window } from '../../shared/vscode/window'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
 import { showOutputMessage } from '../../shared/utilities/messages'
-import { OutputChannel } from 'vscode'
+import { Commands } from '../../shared/vscode/commands'
+import { downloadWithProgress } from '../commands/downloadFileAs'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { readablePath } from '../util'
+import { getStringHash } from '../../shared/utilities/textUtilities'
+import { getLogger } from '../../shared/logger'
+import { showConfirmationMessage } from '../../shared/utilities/messages'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+const SIZE_LIMIT = 4 * Math.pow(10, 6)
+
 export class S3FileViewerManager {
-    //private cache: Set<S3FileNode>
+    private cacheArns: Set<string>
     //private activeTabs: Set<S3Tab>
-    //private window: Window
+    private window: typeof vscode.window
     private outputChannel: OutputChannel
+    private commands: Commands
+    private tempLocation: string | undefined
 
-    public constructor(window: Window = Window.vscode(), outputChannel = ext.outputChannel) {
-        //this.cache = new Set<S3FileNode>()
+    public constructor(
+        cacheArn: Set<string> = new Set<string>(),
+        window: typeof vscode.window = vscode.window,
+        commands = Commands.vscode(),
+        tempLocation?: string
+    ) {
+        this.cacheArns = cacheArn
         //this.activeTabs = new Set<S3Tab>()
```

Comment:
Remove along with the commented-out field? Your comment [here](https://github.com/aws/aws-toolkit-vscode/pull/1863/files#r668107930) says it's not necessary?